src/util/apiUtils.js

```diff
@@ -185,32 +187,44 @@ export const parseApiResponse = requestUrl => ([response, body]) => {
  */
 export const buildRequestArgs = externalRequestOpts =>
 	({ endpoint, params, flags }) => {
-
-		// cheap, brute-force object clone, acceptable for serializable object
-		const externalRequestOptsQuery = JSON.parse(JSON.stringify(externalRequestOpts));
-
-		externalRequestOptsQuery.url = encodeURI(`/${endpoint}`);
-		externalRequestOptsQuery.jar = createCookieJar(externalRequestOptsQuery.url);
+		const dataParams = querystring.stringify(params);
+		const headers = { ...externalRequestOpts.headers };
+		let url = encodeURI(`/${endpoint}`);
+		let body;
+		const jar = createCookieJar(url);
 
 		if (flags) {
-			externalRequestOptsQuery.headers['X-Meetup-Request-Flags'] = flags.join(',');
+			headers['X-Meetup-Request-Flags'] = flags.join(',');
```

Comment:
I'm not sure - I agree with the desire to avoid branching logic, but it also seems wrong to send unnecessary headers, particularly something like this that is framed as a 'request' - it would be like asking for nothing rather than not asking at all, and it essentially shifts the `if` from the client code to the server code - the REST API would have to do the null check on their end to avoid making an empty DB query.

I think I'm happier to keep the client request lean, even at the cost of an `if`